Add branch tracking to development workflow instructions (#28575)
matplotlib/matplotlib: doc/devel/development_workflow.rst

   git push origin my-new-feature

.. hint::

    If you first opened the pull request from your ``main`` branch and then
    converted it to a feature branch, you will need to close the original pull
    request and open a new pull request from the renamed branch. See
    `GitHub: working with branches

>     <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#working-with-branches>`_.

Reviewer: I know you just converted the text to a hint, which is an improvment. But reading this, it feels out of context. We're discussing making a new feature branch. Already having a PR from main means the user has gone two steps in a different direction already and it is not relevant to the feature branch.

I'm slightly in favor of removing this completely. Alternatively, if you think, this is still valuable (Do we expect users to find this when they need it? Is a PR from main really that bad that we need to discuss recovering from this non-standard workflow?), you can move this down to the section "Open a pull request".

Reviewer: Ok let's get this in as an incremental improvement. We can discuss this separately.

    git commit -a --amend --no-edit
    git push [your-remote-repo] [your-branch] --force-with-lease

> .. hint::

Reviewer: ```suggestion
.. tip::
```
IMHO slightly better admontion type (a hint is to make you aware of something, a tip has a touch of  recommendation added).

>     Instead of typying your branch name every time, you can do::

Reviewer: ```suggestion
    Instead of typying your branch name every time, you can once do::
```

>     Instead of typying your branch name every time, you can once do::

Reviewer 2: ```suggestion
    Instead of typing your branch name every time, you can once do::
```

>     Instead of typying your branch name every time, you can once do::

Author: ```suggestion
    Instead of typing your branch name every time, you only need to type the following once to link the remote branch to the local branch::
```
